core/modules/user.py

```diff
@@ -751,28 +750,41 @@ def secondFactorSucceeded(self, secondFactor, userKey):
             raise errors.RequestTimeout()
         return self.authenticateUser(userKey)
 
-    def authenticateUser(self, userKey: db.Key, **kwargs):
+    def authenticateUser(self, key: db.Key, **kwargs):
         """
-            Performs Log-In for the current session and the given userKey.
+            Performs Log-In for the current session and the given user key.
 
             This resets the current session: All fields not explicitly marked as persistent
             by conf["viur.session.persistentFieldsOnLogin"] are gone afterwards.
 
-            :param userKey: The (DB-)Key of the user we shall authenticate
+            :param key: The (DB-)Key of the user we shall authenticate
         """
-        currSess = utils.currentSession.get()
-        res = db.Get(userKey)
-        assert res, "Unable to authenticate unknown user %s" % userKey
-        oldSession = {k: v for k, v in currSess.items()}  # Store all items in the current session
-        currSess.reset()
-        # Copy the persistent fields over
-        for k in conf["viur.session.persistentFieldsOnLogin"]:
-            if k in oldSession:
-                currSess[k] = oldSession[k]
-        del oldSession
-        currSess["user"] = res
-        currSess.markChanged()
-        utils.currentRequest.get().response.headers["Sec-X-ViUR-StaticSKey"] = currSess.staticSecurityKey
+        skel = self.baseSkel()
+        if not skel.fromDB(key):
+            raise ValueError(f"Unable to authenticate unknown user {key}")
+
+        # Update the lastlogin timestamp (if available!)
+        if "lastlogin" in skel:
+            now = utils.utcNow()
+
+            # Conserve DB-Writes: Update the user max once in 30 Minutes (why??)
```

Comment:
In response to the why: the `toDB` triggers updateRelations for `UserBone`s. On every login you would update/write a possible huge amount of entries. 
But we could set `clearUpdateTag=True` in the `toDB` instead. Right?

Reply:
> But we could set `clearUpdateTag=True` in the `toDB` instead. Right?

Right!

Reply:
Then you can remove the 30 min threshold, right?